generator/src/main/java/com/scottlogic/deg/generator/restrictions/linear/LinearRestrictionsFactory.java

```diff
@@ -29,16 +30,66 @@
         return createDateTimeRestrictions(DATETIME_MIN_LIMIT, DATETIME_MAX_LIMIT);
     }
 
-    public static LinearRestrictions<OffsetDateTime> createDateTimeRestrictions(Limit<OffsetDateTime> min, Limit<OffsetDateTime> max) {
+    public static LinearRestrictions<OffsetDateTime> createDateTimeRestrictions(
+        Limit<OffsetDateTime> min,
+        Limit<OffsetDateTime> max) {
+
         return createDateTimeRestrictions(min, max, DEFAULT_DATETIME_GRANULARITY);
     }
 
-    public static LinearRestrictions<OffsetDateTime> createDateTimeRestrictions(Limit<OffsetDateTime> min, Limit<OffsetDateTime> max, Granularity<OffsetDateTime> granularity) {
+    public static LinearRestrictions<OffsetDateTime> createDateTimeRestrictions(
+        Limit<OffsetDateTime> min,
+        Limit<OffsetDateTime> max,
+        Granularity<OffsetDateTime> granularity) {
+
         OffsetDateTime inclusiveMin = getInclusiveMin(min, granularity, ISO_MIN_DATE);
         OffsetDateTime inclusiveMax = getInclusiveMax(max, granularity, ISO_MAX_DATE);
         return new LinearRestrictions<>(inclusiveMin, inclusiveMax, granularity);
     }
 
+    public static LinearRestrictions<LocalTime> createDefaultTimeRestrictions() {
+        return createTimeRestrictions(DEFAULT_TIME_GRANULARITY);
+    }
+
+    public static LinearRestrictions<LocalTime> createTimeRestrictions(Granularity<LocalTime> granularity) {
+        return createTimeRestrictions(TIME_MIN_LIMIT, TIME_MAX_LIMIT, granularity);
+    }
+
+    public static LinearRestrictions<LocalTime> createTimeRestrictions(
+        Limit<LocalTime> min,
+        Limit<LocalTime> max) {
+        return createTimeRestrictions(min, max, DEFAULT_TIME_GRANULARITY);
+    }
+
+
+    /**
+     *This method creates a LocalTime LinearRestriction. If the upper or lower bounds are close to midnight and would
+     * result in a contradictory restrictions, the method ensures that a contradictory LinearRestriction is created.
+     * @param min
+     * @param max
+     * @param granularity
+     * @return
+     */
+    public static LinearRestrictions<LocalTime> createTimeRestrictions(
```

Comment:
I wonder if it would have been better to create a new time restriction in the same fashion as the boolean and string restrictions. That way we don't have to change the behaviour of the linear operator to apply to our specific time implementation, which isn't truly linear.

Reply:
I think this is a valid point. The idea for using linear restrictions was that it would save making another very similar implementation. 

Additionally if a new restrictions class was created, new value sources would also need to be created which would also be very similar.